### ISSUE TITLE
Crash when exchanging blocks

### ISSUE THREAD

**Issue by andr9528** [Direwolf20-MC/BuildingGadgets#461]
**Backstory**
Was exchanging the walls, floors and roof of my new slime spawner, when a big slime spawned and hit me. I'm playing with Mine and Slash, so a level 25 slime vs my level 9, hurts a lot, so i believe i might have gotten one shot, while the blocks were exchanging.

**Versions**
Minecraft: 1.15.2
Forge: 31.1.30
Building Gadgets: 3.3.4

[Crash Report](https://pastebin.com/4znU10M4)

**Reproduction**
I'm am not entirely sure, but maybe dying while blocks are exchanging

TLDR: I think dying while blocks are exchanging does bad things.

**Comment by MichaelHillcox** [comment]
Oh, you died right as the effect block where still animated?

**Comment by andr9528** [comment]
I believe so, trying to open my game again, to see if i am dead. Might be a clash with Hardcore Revival, keeping you in a different state when killed than normal.

**Comment by MichaelHillcox** [comment]
Hmm, is that the mod you use to revive in hardcore? That would be fun to look into. I'll check this one out as it sounds very intresting. Also a great case study for @MajorTuvok on why we don't throw fatal errors because a cap is missing :P

**Comment by andr9528** [comment]
Hardcore Revival is just a mod that is part of the Valhelsia pack i'm playing. Have added a handful of mods to it, including this one. I normally disable that mod, just forgot when i updated yesterday. Hardcore Revival requires you to be 2 players at least to have any effect, it would seem.

On regard to me being dead or alive, i do not know, as i can't re-enter my world now o.O. The exception shown above, is thrown again on world load. I did disable hardcore Revival, so i am trying ones more with it re-enabled. might be really bad exception that i have found. So i might need a quick fix to save my world XD

If it still crashes again i'm going to edit my player data to set alive from false to true, to see if that saves it.

**Comment by MichaelHillcox** [comment]
As you are joining, spam the shit out of a slot number (1-9) in order to change the gadget out of your current slot.

**Comment by MichaelHillcox** [comment]
I've also been able to reproduce this issue, investigating more atm

**Comment by andr9528** [comment]
I don't know if it is a good or bad thing that it was so easy to reproduce D:

I just tried spamming 1 (my sword), while entering, but still end up crashing with the exception.
Going to edit player data to revive me.

**Comment by MichaelHillcox** [comment]
I'll build you a version with it fixed

**Comment by andr9528** [comment]
Last time i check, i think it was in version 1.2.5, there was a boolean, in the player NBT, describing whether or not it was alive, did they change that, or am i miss-remembering?

Also thanks for the help :)

**Comment by MichaelHillcox** [comment]
Running a final test now and if it works you should be able to load back into your world. Honestly, if this issue didn't exist then I would never have found this. Hardcore changes your heath to an invalid number causing some fun issues if the game is intended to be played again :P With this, the player is kinda still alive but not with it's normal stuff so our render is trying to render when it can't.

**Comment by andr9528** [comment]
Fun thing is, my world is not a hardcore world... so does that mean that Hardcore Revival does some bad stuff, even to normal world?

**Comment by MichaelHillcox** [comment]
I've just noticed my code was a little out of date but the issue still exists so I'm just letting my code update. Looks like I have fixed it though. So I'll get you a build as soon as it's done. 

If your world isn't hardcore that means something killed your player in a really bad way. Something hacky I guess because when your player dies it should remove the entitiy from the world and when you respawn it should set you back up. For some reason this didn't happen and you kinda sudo died meaning we could still access your player entite but we couldn't access your inventory. This is an odd one but fixable as we shouldn't be crashing the game if we can't find your inventory. That bit is completely on us.

**Comment by andr9528** [comment]
Could i be Corpse then, leaving behind, well, my corpse? 

It leaves behind a corpse with your items in it. 
For me it is just a empty corpse as i do play with KeepInventory = True.

It might still be Hardcore Revival, as it keeps you in a not dead but not alive state for up to 2 min after you "die", allowing other player to revive you. You are simply downed, incapable of doing anything. After a 5 sec delay, the player can click "Give up" or something like that, at which point they die for real.

**Comment by MichaelHillcox** [comment]
I'll check out there mod to see what they're doing on the death event. 

https://s.mikey.pro/buildinggadgets-3.3.4.jar

Please try and launch your world with that version instead. I've not bumped the version number so it should just ask you to replace it. Let me know if this fixes it or if you get a different crash

**Comment by MichaelHillcox** [comment]
https://github.com/blay09/HardcoreRevival/blob/1.15.x/src/main/java/net/blay09/mods/hardcorerevival/handler/DeathHandler.java#L81 Looks like they hold the player just before the game would have natrually invalidated the data. So I don't know what's going on haha. The above should solve / bypass the issue though :D

**Comment by andr9528** [comment]
When overwriting the current version i had, with the one supplied by you, it somehow just deleted both files. This i didn't notice until after i had loaded my world, so good news removing building gadgets fixes the exception, who would have known XD.
It did show the warning that stuff was missing, but i didn't pay much attention to it, so that's on me. With regards to me losing the gadgets, by removing the mod, don't worry i only had the exchanging gadgets, so it is easy to cheat back in. When i finally loaded up the game with the version supplied by you, it worked just fine. Thanks for all the assistance.

With regard to the level 25 slime, it did in fact one shot me.

**Comment by andr9528** [comment]
Fun fact: each time i had tried logging in, while the exception was there, Journey Map took that as another death, adding a way point.
